### DISCUSSION TITLE
Is there a way to make the extension delete the CMakeCache.txt when switching presets?

### DISCUSSION THREAD

**Discussion by gabeklavans** [microsoft/vscode-cmake-tools#3333] [Q&A] [answered]
I'm using CMake Presets (with a CMakePresets.json file) and I have a preset configuration for building on the host arch and one for cross-compiling to Android using the NDK. Currently, if I just switch presets, certain cache variables like `CMAKE_C_COMPILER` remain unchanged, when they should be set using the toolchain file to a cross-compilation compiler. This means I have to manually delete the CMakeCache.txt file whenever I want to switch between these presets.

Is there a way to have the extension just delete the cache file for me when I switch configs?

Thanks

**Comment by gcampbell-msft** [comment, marked as answer]
There isn't anything explicit in our extension that, by default, deletes the cache file fully when you switch configurations. However, we do have a setting now that, upon a "Clean Configure", we ensure to delete the entire build directory. `cmake.deleteBuildDirOnCleanConfigure`. 

You could set this and then do a clean configure rather than having to manually delete things.
